tests/uart-loopback/src/main.rs

```diff
@@ -0,0 +1,53 @@
+//! This is a test for UART loopback operation
+
+#![no_main]
+#![no_std]
+
+mod pins;
+
+use ariel_os::{
+    debug::{
+        ExitCode, exit,
+        log::{Hex, info},
+    },
+    hal,
+    time::Duration,
+    time::with_timeout,
+    uart::Baud,
+};
+
+use embedded_io_async::{Read as _, Write as _};
+
+#[ariel_os::task(autostart, peripherals)]
+async fn main(peripherals: pins::Peripherals) {
+    info!("Starting UART test");
+
+    let mut config = hal::uart::Config::default();
+    config.baudrate = Baud::_9600;
+    info!("Selected configuration: {}", config);
+
+    let mut rx_buf = [0u8; 32];
+    let mut tx_buf = [0u8; 32];
+
+    let mut uart = pins::TestUart::new(
+        peripherals.uart_rx,
+        peripherals.uart_tx,
+        &mut rx_buf,
+        &mut tx_buf,
+        config,
+    );
+
+    const OUT: &str = "Test Message";
+    let mut in_ = [0u8; OUT.len()];
```

Comment:
There's other ways to name that variables to not have to resort to using an underscore:

```suggestion
    let mut input = [0u8; OUT.len()];
```

Reply:
Copied from the SPI loopback test :innocent:, fixed in c94fcfd58b1f7aa0b1b7b880e48c7e4452c8f169. Let me know if you find it worth the effort to also rename the variable in the spi loopback test :smile: